src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDataTable.java

```diff
@@ -89,11 +93,30 @@ public synchronized void addColumnMetadata(String columnName,
      */
     public synchronized void addColumnMetadata(SQLServerDataColumn column) throws SQLServerException {
         // column names must be unique
-        Util.checkDuplicateColumnName(column.columnName, columnMetadata);
+        checkDuplicateColumnName(column.columnName);
         columnMetadata.put(columnCount++, column);
     }
 
     /**
+     * Checks if duplicate columns exists, in O(n) time.
+     * 
+     * @param columnName
+     *            the name of the column
+     * @throws SQLServerException
+     *             when a duplicate column exists
+     */
+    private void checkDuplicateColumnName(String columnName) throws SQLServerException {
+        if (null != columnList) {
```

Comment:
I'd rather an NPE than silently fail to check dupes if someone messes up null handling.

Better IMO would be defining the local variable like `Set<String> columnNames = new HashSet<>();` and omitting this check, but I don't know if that'd violate the project's coding style.